jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -0,0 +1,118 @@
+from glue.core.message import EditSubsetMessage, SubsetUpdateMessage
+from glue.core.edit_subset_mode import (AndMode, AndNotMode, OrMode,
+                                        ReplaceMode, XorMode)
+from glue.core.subset import (RoiSubsetState, RangeSubsetState,
+                              OrState, AndState, XorState, InvertState)
+from glue_jupyter.widgets.subset_mode_vuetify import SelectionModeMenu
+from traitlets import List, Unicode, Bool, Dict, observe
+
+from jdaviz.core.registries import tray_registry
+from jdaviz.core.template_mixin import TemplateMixin, SubsetSelect
+
+__all__ = ['SubsetPlugin']
+
+SUBSET_MODES = {
+    'replace': ReplaceMode,
+    'add': OrMode,
+    'and': AndMode,
+    'xor': XorMode,
+    'remove': AndNotMode
+}
+
+
+@tray_registry('g-subset-plugin', label="Subset Tools")
+class SubsetPlugin(TemplateMixin):
+    template_file = __file__, "subset_plugin.vue"
+    select = List([]).tag(sync=True)
+    subset_items = List([]).tag(sync=True)
+    subset_selected = Unicode("Create new").tag(sync=True)
+    mode_selected = Unicode('add').tag(sync=True)
+    show_region_info = Bool(False).tag(sync=True)
+    subset_classname = Unicode('').tag(sync=True)
+    subset_definition = Dict({}).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.components = {
+            'g-subset-mode': SelectionModeMenu(session=self.session)
+        }
+
+        self.session.hub.subscribe(self, EditSubsetMessage,
+                                   handler=self._sync_selected_from_state)
+        self.session.hub.subscribe(self, SubsetUpdateMessage,
+                                   handler=self._get_region_definition)
+
+        self.no_selection_text = "Create new"
+        self.subset_select = SubsetSelect(self,
+                                          'subset_items',
+                                          'subset_selected',
+                                          default_text=self.no_selection_text)
+
+    def _sync_selected_from_state(self, *args):
+        if self.session.edit_subset_mode.edit_subset == []:
+            if self.subset_selected != self.no_selection_text:
+                self.subset_selected = self.no_selection_text
+                self.show_region_info = False
+        else:
+            new_label = self.session.edit_subset_mode.edit_subset[0].label
+            if new_label != self.subset_selected:
+                if new_label not in [s['label'] for s in self.subset_items]:
+                    self._sync_available_from_state()
```

Comment:
I think this is likely where the spectral vs spatial confusion is happening.  This is forcing the entry in the choices before the existing logic is ready to detect the type 🤔